diff -ru work/kernel/nvidia/nv-pat.c work2/kernel/nvidia/nv-pat.c
--- work/kernel/nvidia/nv-pat.c	2020-07-06 00:40:53.000000000 +0900
+++ work2/kernel/nvidia/nv-pat.c	2020-08-05 10:15:29.639409164 +0900
@@ -51,14 +51,14 @@
     wbinvd();
     *cr4 = NV_READ_CR4();
     if (*cr4 & 0x80) NV_WRITE_CR4(*cr4 & ~0x80);
-    __flush_tlb();
+    __flush_tlb_local();
 }
 
 static inline void nv_enable_caches(unsigned long cr4)
 {
     unsigned long cr0 = read_cr0();
     wbinvd();
-    __flush_tlb();
+    __flush_tlb_local();
     write_cr0((cr0 & 0x9fffffff));
     if (cr4 & 0x80) NV_WRITE_CR4(cr4);
 }
