--- /etc/crontab	2013-12-31 08:13:54.352676704 +0900
+++ crontab	2014-07-07 15:51:18.395217585 +0900
@@ -1,5 +1,4 @@
 # for vixie cron
-# $Header: /var/cvsroot/gentoo-x86/sys-process/vixie-cron/files/crontab-3.0.1-r4,v 1.3 2011/09/20 15:13:51 idl0r Exp $
 
 # Global variables
 SHELL=/bin/bash
@@ -9,7 +8,7 @@
 
 # check scripts in cron.hourly, cron.daily, cron.weekly and cron.monthly
 59  *  * * *	root	rm -f /var/spool/cron/lastrun/cron.hourly
-9  3  * * *	root	rm -f /var/spool/cron/lastrun/cron.daily
-19 4  * * 6	root	rm -f /var/spool/cron/lastrun/cron.weekly
+9  22  * * *	root	rm -f /var/spool/cron/lastrun/cron.daily
+19 23  * * 6	root	rm -f /var/spool/cron/lastrun/cron.weekly
 29 5  1 * *	root	rm -f /var/spool/cron/lastrun/cron.monthly
 */10  *  * * *	root	test -x /usr/sbin/run-crons && /usr/sbin/run-crons
