--- source/va.x86_64/etc/crontab	2016-04-03 19:24:44.950006167 +0900
+++ /tmp/crontab	2016-10-03 14:18:31.055127800 +0900
@@ -9,7 +9,7 @@
 
 # check scripts in cron.hourly, cron.daily, cron.weekly and cron.monthly
 59  *  * * *	root	rm -f /var/spool/cron/lastrun/cron.hourly
-9  3  * * *	root	rm -f /var/spool/cron/lastrun/cron.daily
-19 4  * * 6	root	rm -f /var/spool/cron/lastrun/cron.weekly
+9  22  * * *	root	rm -f /var/spool/cron/lastrun/cron.daily
+19 23  * * 6	root	rm -f /var/spool/cron/lastrun/cron.weekly
 29 5  1 * *	root	rm -f /var/spool/cron/lastrun/cron.monthly
 */10  *  * * *	root	test -x /usr/sbin/run-crons && /usr/sbin/run-crons
