reverted:
--- b/arch/x86/include/asm/i8259.h
+++ a/arch/x86/include/asm/i8259.h
@@ -60,7 +60,6 @@
 	void (*mask_all)(void);
 	void (*restore_mask)(void);
 	void (*init)(int auto_eoi);
-	int (*probe)(void);
 	int (*irq_pending)(unsigned int irq);
 	void (*make_irq)(unsigned int irq);
 };
reverted:
--- b/arch/x86/kernel/apic/vector.c
+++ a/arch/x86/kernel/apic/vector.c
@@ -361,11 +361,7 @@
 	if (nr < nr_irqs)
 		nr_irqs = nr;
 
+	return nr_legacy_irqs();
-	/*
-	 * We don't know if PIC is present at this point so we need to do
-	 * probe() to get the right number of legacy IRQs.
-	 */
-	return legacy_pic->probe();
 }
 
 #ifdef	CONFIG_X86_IO_APIC
reverted:
--- b/arch/x86/kernel/i8259.c
+++ a/arch/x86/kernel/i8259.c
@@ -295,11 +295,16 @@
 	raw_spin_unlock_irqrestore(&i8259A_lock, flags);
 }
 
+static void init_8259A(int auto_eoi)
-static int probe_8259A(void)
 {
 	unsigned long flags;
 	unsigned char probe_val = ~(1 << PIC_CASCADE_IR);
 	unsigned char new_val;
+
+	i8259A_auto_eoi = auto_eoi;
+
+	raw_spin_lock_irqsave(&i8259A_lock, flags);
+
 	/*
 	 * Check to see if we have a PIC.
 	 * Mask all except the cascade and read
@@ -307,28 +312,16 @@
 	 * have a PIC, we will read 0xff as opposed to the
 	 * value we wrote.
 	 */
-	raw_spin_lock_irqsave(&i8259A_lock, flags);
-
 	outb(0xff, PIC_SLAVE_IMR);	/* mask all of 8259A-2 */
 	outb(probe_val, PIC_MASTER_IMR);
 	new_val = inb(PIC_MASTER_IMR);
 	if (new_val != probe_val) {
 		printk(KERN_INFO "Using NULL legacy PIC\n");
 		legacy_pic = &null_legacy_pic;
+		raw_spin_unlock_irqrestore(&i8259A_lock, flags);
+		return;
 	}
 
-	raw_spin_unlock_irqrestore(&i8259A_lock, flags);
-	return nr_legacy_irqs();
-}
-
-static void init_8259A(int auto_eoi)
-{
-	unsigned long flags;
-
-	i8259A_auto_eoi = auto_eoi;
-
-	raw_spin_lock_irqsave(&i8259A_lock, flags);
-
 	outb(0xff, PIC_MASTER_IMR);	/* mask all of 8259A-1 */
 
 	/*
@@ -386,10 +379,6 @@
 {
 	return 0;
 }
-static int legacy_pic_probe(void)
-{
-	return 0;
-}
 
 struct legacy_pic null_legacy_pic = {
 	.nr_legacy_irqs = 0,
@@ -399,7 +388,6 @@
 	.mask_all = legacy_pic_noop,
 	.restore_mask = legacy_pic_noop,
 	.init = legacy_pic_int_noop,
-	.probe = legacy_pic_probe,
 	.irq_pending = legacy_pic_irq_pending_noop,
 	.make_irq = legacy_pic_uint_noop,
 };
@@ -412,7 +400,6 @@
 	.mask_all = mask_8259A,
 	.restore_mask = unmask_8259A,
 	.init = init_8259A,
-	.probe = probe_8259A,
 	.irq_pending = i8259A_irq_pending,
 	.make_irq = make_8259A_irq,
 };
