--- linux-4.9.5-aufs/fs/aufs/branch.c.orig	2017-01-27 09:30:29.423491472 +0900
+++ linux-4.9.5-aufs/fs/aufs/branch.c	2017-01-27 09:24:23.582978707 +0900
@@ -250,7 +250,7 @@
 		goto out;
 	}
 
-	if (unlikely(inode->i_sb->s_stack_depth)) {
+	if (!inode->i_sb->s_stack_depth && strcmp(au_sbtype(inode->i_sb), "overlay")) {
 		pr_err("already stacked, %s (%s)\n",
 		       add->pathname, au_sbtype(inode->i_sb));
 		goto out;
